src/components/companies/detail/overview/CompanyOverview.tsx

```diff
@@ -90,27 +105,90 @@ export function CompanyOverview({
       )
     : t("companies.overview.notReported");
 
+  const handleAssessEmissions = async () => {
+    setIsLoadingAssessment(true);
+    setAssessmentError(null);
+    try {
+      const response = await fetch('/api/emissions-assessment', {
```

Comment:
We use [React Query](https://tanstack.com/query/v4/docs/framework/react/overview) in the frontend to handle the fetching and caching of API calls. A good pattern is to create a custom hook that encapsulates this.

That way you can combine the various states into:
```typescript
const {loading, error, emissionAssessment} = useEmissionAssessment(company.wikidataId, selectedYears)
```

You can look at useCompanies as an example for how to do that.